src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -948,6 +948,8 @@ private static uint GetParameterRefEscape(ParameterSymbol parameter)
             {
                 { RefKind: RefKind.None } => Binder.CurrentMethodScope,
                 { EffectiveScope: DeclarationScope.RefScoped } => Binder.CurrentMethodScope,
+                { RefKind: RefKind.Out } => Binder.ReturnOnlyScope,
```

Comment:
Consider including some redundant tests in these patterns to improve clarity here, knowing that they won't be emitted.

Reply:
Thanks for the suggestion. Added `HasUnscopedRefAttribute: true` to this case.

And I think if we use `DeclaredScoped` rather than `EffectiveScope` in the previous case, it would improve clarity there as well, but I've left that for [https://github.com/dotnet/roslyn/issues/65293](https://github.com/dotnet/roslyn/issues/65293).